src/telemetry/Manager.cc

```diff
@@ -160,6 +155,16 @@ void Manager::InitPostScript() {
     iosource_mgr->RegisterFd(collector_flare->FD(), this);
 }
 
+void Manager::Terminate() {
+    // Shut down the exposer first of all so we stop getting requests for
+    // data. This keeps us from getting a request on another thread while
+    // we're shutting down.
```

Comment:
I believe this should have the `collector_cv.notify_all()` call here, otherwise I saw a 5 second shutdown hang once in a while when running a `while true;  do curl <metrics endpoint>; done`. It terminated cleanly, but essentially the threads were waiting for the 5 seconds timeout, just to find that the predicate is now fulfilled as Zeek is terminating.

It's basically the `.reset()` and `destructor` waiting for all requests to finish.